metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -263,6 +265,14 @@ protected TxnStore initialValue() {
       }
     };
 
+    private static final ThreadLocal<Map<String, com.codahale.metrics.Timer.Context>> timerContexts =
```

Comment:
Checkout CodahaleMetrics.java line 99 in the current code.  It is doing the same things.  In both cases, the Timer is not ThreadLocal, the Context is.  So in each case we can get average time across all instances (as you say).  But we have to keep the Context thread local because this represents on entry into and exit from the function by a particular thread.

Reply:
got it ..Thanks for pointer.